Feat/change item values

starushkja opened the pull request.

starushkja committed:
feat/change the content of row
* copy values from row to the input's form on doubleclick

starushkja committed:
* add buttons "save" and "cancel"
	*add function to save changes in input's form
	* add function to cancel all changes in input's form

starushkja committed:
feat/change-buttons-features
*buttons"save" and"cancel" invisible, they appear after dblclick row, button "add" removes
*after finish of edit and press button "save" or "cancel", button
"add" reset, "save and "cancel" are invisible
* buttons "delete", "up" , "down" are disabled in edit mode

checkpoint20 commented on designs.js:
```diff
@@ -2,15 +2,21 @@ const listTitle = document.getElementById('inputTitle');
+const saveButton = document.getElementById("saveButton");
+const clearButton = document.getElementById("clearButton");
+const inputDelete = document.getElementById("inputDelete");
+const up = document.getElementById("up");
```
I would recommend to normalize naming of these constants. It helps when all names follow same pattern. Eg. "{action}{Element}" like "saveButton". Otherwise it makes reading code more difficult. The same is true for ids. In my view two lines below look great. All others should be changed in the same way.
```
const saveButton = document.getElementById("saveButton");
const clearButton = document.getElementById("clearButton");
```

starushkja committed:
feat/change the name of buttons to the "buttonDelete", "buttonUp",
"buttonDown"

checkpoint20 commented on designs.js:
```diff
@@ -2,15 +2,21 @@ const listTitle = document.getElementById('inputTitle');
-  }
-)
+submit.addEventListener('submit', makeList);
+clearForm();
```
These naming looks inconsistent. Some buttons are buttonAction, some actionButton. Could you please fix this. I would also suggest to rename html element ids respectively.

starushkja committed:
feat/improwment of previous functions and creation the new function
*make function "cancel",which starts after onckick button "cancel" and
abolishes changes
*make function "contraEdit" to escape repetition of the same in
functions

checkpoint20 commented on designs.js:
```diff
@@ -34,15 +34,11 @@ function makeList(event){
 }
 
 function copyDataToAddForm(event){
+    event.preventDefault();
```
Same thing as line 54.
We tried to use event.preventDefault to solve the issue with form submit but you've solved it differently, so we don't need it here.

checkpoint20 commented on designs.js:
```diff
@@ -55,16 +51,16 @@ function selectRowHandler(event){
 }
 
-function save(){
+function save(event){
```
Could you please remove event from the argument list? It looks like we don't use it in the function.

checkpoint20 commented on designs.js:
```diff
@@ -55,16 +51,16 @@ function selectRowHandler(event){
+    clearForm();
+}
+
+function cancel(event){
```
Same thing as line 54

checkpoint20 reviewed (commented):
This change looks really good. Only two minors:
1. We can remove event form function arguments, please check my comments in code.
2. When you have several commits in one pool request it's better to keep same header in commit messages.
